frontend/.eslintrc.js

```diff
@@ -0,0 +1,170 @@
+
+    // Yoda style, equality comparisons with literals must be.
+    "yoda": [ "error", "always", { "onlyEquality" : true } ],
+
+    // No type coercion in comparisons.
+    "eqeqeq": ["error", "always" ],
+
+    // No declaring anything and not using it; did you finish cleaning up?
+    "no-unused-vars": ["error", { "args": "none" }],
+
+    // Must use const or let.
+    "no-var": "error",
+
+    // Stops unknown globals.
+    "no-undef": "error",
+
+    // Arrow functions instead of classic function syntax. https://stackoverflow.com/a/64258560/5301718
+    "prefer-arrow/prefer-arrow-functions": [
+      "error",
+      {
+        "disallowPrototype": true,
+        "singleReturnOnly": false,
+        "classPropertiesAllowed": false
+      }
+    ],
+    "prefer-arrow-callback": [
+      "error",
+      { "allowNamedFunctions": true }
+    ],
+    // Disabled for now, because this seems to be redundant, but it might not be in some cases; preserved for if we find those cases.
+    //"func-style": [
+    //  "error",
+    //  "expression"
+    //],
+
+
+    // React.
+    // Prevent React to be incorrectly marked as unused.
+    "react/jsx-uses-react": "error",
+    "react/jsx-uses-vars": "error",
+    // No inline styles. 2 is the same as "error" for this plugin. https://www.npmjs.com/package/eslint-plugin-no-inline-styles
+    "no-inline-styles/no-inline-styles": 2,
+
+
+    // JSDoc
+    // Require JSDoc block.
+    "jsdoc/require-jsdoc": "error",
+    // Require a description in JSDoc block.
+    "jsdoc/require-description": [
+      "error",
+      {
+        "descriptionStyle": "any",
+        "checkConstructors": false,
+        "checkGetters": false,
+        "checkSetters": false
+      }
+    ],
+    "jsdoc/newline-after-description": "error",
+    // Require a return type. Don't require a return type description.
+    "jsdoc/require-returns": "error",
+    "jsdoc/require-returns-type": "error",
+    "jsdoc/require-returns-description": "off",
+    // Params need a type and description.
+    "jsdoc/check-types": "error",
+    "jsdoc/require-param-description": "error",
+
+
+    // Jest
+    // Tests must have at least one assertion.
+    "jest/expect-expect": "error",
+
+  },
+  overrides: [
+    {
+      // Separate linting for typescript. https://stackoverflow.com/a/60773716/5301718
+      files: ["*.ts", "*.tsx"],
+      parser: "@typescript-eslint/parser",
+      plugins: [
+        "@typescript-eslint",
+        "eslint-plugin-no-inline-styles",
+        "prefer-arrow"
+      ],
+      extends: [
+        "plugin:@typescript-eslint/eslint-recommended",
+      ],
+      parserOptions: {
+        "ecmaFeatures": {
+          "jsx": true
+        },
+        "ecmaVersion": "latest",
+        "sourceType": "module"
+      },
+      settings: {
+        react: {
+          version: '17',
+        },
+      },
+
+      /**
+       * Typescript Rules
+       */
+      rules: {
+        "@typescript-eslint/no-explicit-any": "off",
```

Comment:
I know this isn't in the acceptance criteria but I would have liked to see enforcement of "no explicit any" and then putting in a linter "ignore" statement if one is necessary. That said, I don't know to what degree you guys are committing to Typescript with the PMO app and maybe allowing "any" is what you're going with.

Reply:
this thought crossed my mind at one point as well. if we both had it, its probably good to do, so i will implement this.